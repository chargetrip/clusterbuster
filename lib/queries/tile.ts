import { GetTileQuery, ITileQuery } from '../types/index';

/**
 * @description The default tile query builder
 */
export const defaultGetTileQuery: GetTileQuery = ({
  x,
  y,
  z,
  table,
  geometry,
  extent,
  bufferSize,
  attributes,
}: ITileQuery) => `
SELECT
  ST_AsMVTGeom(ST_Transform(${geometry}, 3857), TileBBox(${z}, ${x}, ${y}, 3857), ${extent}, ${bufferSize}, false) AS geom,
  jsonb_build_object(
    'count', size, 
    'expansionZoom', expansionZoom, 
    'lng', ST_X(ST_Transform(${geometry}, 4326)), 
    'lat', ST_Y(ST_Transform(${geometry}, 4326))${attributes}
  ) AS attributes
FROM ${table}
`;
