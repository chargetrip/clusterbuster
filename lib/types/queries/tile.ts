/**
 * @description Input interface for the tile query builder
 */
export interface ITileQuery {
  x: number;
  y: number;
  z: number;
  table: string;
  geometry: string;
  extent: number;
  bufferSize: number;
  attributes: string;
}

/**
 * @description The tile query builder callback definition
 */
export type GetTileQuery = (input: ITileQuery) => string;
