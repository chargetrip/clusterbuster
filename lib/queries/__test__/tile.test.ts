import { defaultGetTileQuery } from '../tile';

describe('defaultGetTileQuery', () => {
  it('should work', () => {
    expect(
      defaultGetTileQuery({
        x: 0,
        y: 0,
        z: 0,
        table: 'table',
        geometry: 'geometry',
        extent: 4096,
        bufferSize: 256,
        attributes: ", 'id', id, 'name', name",
      })
    ).toMatchSnapshot();
  });
});
