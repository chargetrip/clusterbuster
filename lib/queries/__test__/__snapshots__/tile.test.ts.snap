// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultGetTileQuery should work 1`] = `
"
SELECT
  ST_AsMVTGeom(ST_Transform(geometry, 3857), TileBBox(0, 0, 0, 3857), 4096, 256, false) AS geom,
  jsonb_build_object(
    'count', size, 
    'expansionZoom', expansionZoom, 
    'lng', ST_X(ST_Transform(geometry, 4326)), 
    'lat', ST_Y(ST_Transform(geometry, 4326)), 'id', id, 'name', name
  ) AS attributes
FROM table
"
`;
